lib/text/listener.js

```diff
@@ -83,6 +83,82 @@ class Importer extends SHRParserListener {
     }
   }
 
+    enterValuesetDefsDoc(ctx) {
+        const ns = ctx.valuesetDefsHeader().namespace().getText();
+        this._currentNs = ns;
+        if (typeof this._nsMap[ns] == 'undefined') {
+            this._nsMap[ns] = new Namespace(ns);
```

Comment:
I think you need to keep a separate map for the ValueSet definitions, otherwise you're mixing your valueset definitions with your element definitions (since this `Importer` class iterates over many files!).